packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -208,7 +208,10 @@ function hydrateComment(node: Node, vnode: VComment, renderer: RendererAPI): Nod
 }
 
 function hydrateStaticElement(elm: Node, vnode: VStatic, renderer: RendererAPI): Node | null {
-    if (!areCompatibleNodes(vnode.fragment, elm, vnode, renderer)) {
+    if (
+        !hasCorrectNodeType<Element>(vnode, elm, EnvNodeTypes.ELEMENT, renderer) ||
+        !areCompatibleNodes(vnode.fragment, elm, vnode, renderer)
+    ) {
```

Comment:
Needed because of TypeScript issues, have to ensure `elm` is an `Element` is an `elm` since `VStatic`'s `elm` is now a stricter `Element` rather than `Node`. This same check is done in `hydrateElement`.